src/vidpf.rs

```diff
@@ -366,8 +381,48 @@ impl VidpfKey {
     }
 }
 
+impl Decode for VidpfKey {
+    fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        let id = u8::decode(bytes)?;
+        let mut value = [0; 16];
+        bytes.read_exact(&mut value)?;
+        Ok(VidpfKey {
+            id: match id {
+                0 => VidpfServerId::S0,
+                1 => VidpfServerId::S1,
+                _ => return Err(CodecError::UnexpectedValue),
+            },
+            value,
+        })
+    }
+}
+
+impl Encode for VidpfKey {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        bytes.push(match self.id {
+            VidpfServerId::S0 => 0u8,
+            VidpfServerId::S1 => 1u8,
+        });
+        bytes.extend_from_slice(&self.value[..]);
+        Ok(())
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        Some(17)
+    }
+}
+
+impl PartialEq for VidpfKey {
+    fn eq(&self, other: &VidpfKey) -> bool {
+        if self.id == other.id {
+            return self.value == other.value;
+        };
+        false
+    }
+}
```

Comment:
This implementation is incomplete because it doesn't compare the key bytes themselves.

Where do we need `PartialEq` for `VidpfKey`?

Reply:
The VDAF interface requires PartialEq for MasticInputShare, which contains a VidpfKey.